lib/ziti-tunnel/ziti_tunnel.c

```diff
@@ -550,6 +559,62 @@ static void run_packet_loop(uv_loop_t *loop, tunneler_context tnlr_ctx) {
     uv_timer_start(&tnlr_ctx->lwip_timer_req, check_lwip_timeouts, 0, 10);
 }
 
+typedef struct ziti_tunnel_async_call_s {
+    ziti_tunnel_async_fn f;
+    void *               arg;
+} ziti_tunnel_async_call_t;
+
+static void async_close_cb(uv_handle_t *async) {
+    if (async->data != NULL) {
+        free(async->data);
+    }
+    free(async);
+}
+
+/** invoke a caller-supplied function with argument. this is called by libuv on the loop thread */
+static void ziti_tunnel_async_wrapper(uv_async_t *async) {
+    ziti_tunnel_async_call_t *call = async->data;
+    if (call != NULL && call->f != NULL) {
+        call->f(async->loop, call->arg);
+    }
+    uv_close((uv_handle_t *)async, async_close_cb);
+}
+
+/** sets up a function call on the specified loop */
+void ziti_tunnel_async_send(tunneler_context tctx, ziti_tunnel_async_fn f, void *arg) {
+    uv_loop_t *loop = uv_default_loop();
+    if (tctx != NULL) {
+        loop = tctx->loop;
+    }
+
+    ziti_tunnel_async_call_t *call = calloc(1, sizeof(ziti_tunnel_async_call_t));
+    call->f = f;
+    call->arg = arg;
+
+    uv_async_t *async = calloc(1, sizeof(uv_async_t));
+    async->data = call;
+
+    if (tctx != NULL) {
```

Comment:
could re go back to the version of this function that had pointer variables for the loop and semaphore objects?

Reply:
I added the old code back, please check